pkg/csi/service/wcpguest/controller.go

```diff
@@ -199,13 +237,24 @@ func (c *controller) ReloadConfiguration() {
 func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (
 	*csi.CreateVolumeResponse, error) {
 
+	start := time.Now()
+	isFileVolumeRequest := common.IsFileVolumeRequest(ctx, req.GetVolumeCapabilities())
+	var volumeType string
+	if isFileVolumeRequest {
+		volumeType = common.PrometheusFileVolumeType
+	} else {
+		volumeType = common.PrometheusBlockVolumeType
+	}
+
 	ctx = logger.NewContextWithLogger(ctx)
 	log := logger.GetLogger(ctx)
 	log.Infof("CreateVolume: called with args %+v", *req)
 	err := validateGuestClusterCreateVolumeRequest(ctx, req)
 	if err != nil {
 		msg := fmt.Sprintf("Validation for CreateVolume Request: %+v has failed. Error: %+v", *req, err)
 		log.Error(msg)
+		volumeControlOpsHistVec.WithLabelValues(volumeType, common.PrometheusCreateOpType,
```

Comment:
Instead of making sure to scrape every failure, Can we simplify this as we did with in-tree VCP?
this way in the future anyone updating code block will never need to worry about if they are handling and reporting failure to Prometheus.

https://github.com/kubernetes/kubernetes/blob/661eae7c7c8226a3f7bed3a24acf1c22861477b7/staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go#L1496-L1498

```
        requestTime := time.Now()
	err := deleteVolumeInternal(vmDiskPath)
	vclib.RecordvSphereMetric(vclib.OperationDeleteVolume, requestTime, err)
```